### DISCUSSION TITLE
what are the major differences in between MXGraph and MAXGraph

### DISCUSSION THREAD

**Discussion by Venkat5598432** [maxGraph/maxGraph#515] [Q&A] [answered]
Please share the major differences in between MXGraph and MAXGraph.

I know one difference. MXGraph is javascript library and MAXGrapgh is TypeScript Library.  So Please let me know the same any other few more differences.

**Comment by tbouffard** [comment, marked as answer]
Hi, here are some insights.
Remember that maxGraph is a fork of mxGraph.
See also https://github.com/maxGraph/maxGraph/blob/v0.13.0/README.md#history

Here are five major differences between **mxGraph** and **maxGraph**:

- **Development and Maintenance**:
  - **mxGraph**: Development of mxGraph was discontinued in 2020, and it is no longer actively maintained.
  - **maxGraph**: maxGraph is an actively maintained fork of mxGraph, with ongoing updates and improvements.

- **TypeScript Support**:
  - **mxGraph**: Written primarily in JavaScript, with no native TypeScript support (requires external dependency: https://github.com/typed-mxgraph/typed-mxgraph).
  - **maxGraph**: Fully converted to TypeScript, providing enhanced type safety, a better development experience, and easier integration with TypeScript projects.

- **Community and Contributions**:
  - **mxGraph**: With its discontinuation, there are no ongoing contributions or updates from the community.
  - **maxGraph**: Actively maintained with a focus on community contributions, ensuring the project continues to evolve and improve based on user feedback.

- **Compatibility and Modern Codebase**:
  - **mxGraph**: Designed to be compatible with older web technologies, including support for legacy browsers like Internet Explorer 8 and VML (Vector Markup Language).
  - **maxGraph**: Features a more modern codebase that aligns with current web development standards, making it better suited for contemporary browsers and projects.

- **Tree-Shaking**:
  - **mxGraph**: Lacks built-in support for tree-shaking, which may result in larger bundle sizes.
  - **maxGraph**: Supports tree-shaking, allowing developers to optimize their applications by including only the parts of the library they need, leading to smaller and more efficient bundles.
